__tests__/cron/exportToTinybird.ts

```diff
@@ -0,0 +1,345 @@
+import { DataSource } from 'typeorm';
+import { saveFixtures } from '../helpers';
+import createOrGetConnection from '../../src/db';
+import { ArticlePost, PostTag, Source, User } from '../../src/entity';
+import { sourcesFixture } from '../fixture/source';
+import { postsFixture, postTagsFixture } from '../fixture/post';
+import {
+  PostsMetadataRepositoryDependency,
+  PostsRepositoryDependency,
+  PostsMetadataRepository,
+  PostsRepository,
+  TinybirdExportService,
+  TinybirdPost,
+} from '../../src/cron/exportToTinybird';
+import {
+  ITinybirdClient,
+  TinybirdDatasourceMode,
+  PostDatasourceResult,
+  QueryResult,
+  fetchfn,
+  TinybirdClient,
+} from '../../src/common/tinybird';
+import { FastifyBaseLogger } from 'fastify';
+
+let con: DataSource;
+
+beforeAll(async () => {
+  con = await createOrGetConnection();
+});
+
+beforeEach(async () => {
+  jest.clearAllMocks();
+
+  await saveFixtures(con, Source, sourcesFixture);
+  await saveFixtures(con, ArticlePost, postsFixture);
+  await saveFixtures(con, PostTag, postTagsFixture);
+  await con
+    .getRepository(User)
+    .save({ id: '1', name: 'Ido', image: 'https://daily.dev/ido.jpg' });
+  await con.getRepository(User).save({
+    id: '2',
+    name: 'Lee',
+    image: 'https://daily.dev/lee.jpg',
+  });
+});
+
+describe('PostsRepository', () => {
+  it('should return posts to export to tinybird with specific properties', async () => {
+    const now = new Date();
+    const latest = new Date(now.getTime() - 10000);
+    const repo = new PostsRepository(con);
+
+    const posts = await repo.getForTinybirdExport(latest);
+    posts.forEach((post) => {
+      post.created_at = '';
+      post.metadata_changed_at = '';
+    });
+    expect(posts).toMatchSnapshot();
```

Comment:
We had some issues in the past where the order of the snapshot can change, thus the inline object test was a better solution.
Would probably be best to revert to that to avoid flaky order issues.

Reply:
Done